tests/integration/sync/test_sync_command.py

```diff
@@ -0,0 +1,398 @@
+        self.assertIn("Error: Missing option '--stack-name'.", str(sync_process_execute.stderr))
+
+    @parameterized.expand(["template.yaml"])
+    def test_sync_infra_no_capabilities(self, template_file):
+        template_path = str(self.test_data_path.joinpath(template_file))
+        stack_name = self._method_to_stack_name(self.id())
+        self.stacks.append({"name": stack_name})
+
+        # Run infra sync
+        sync_command_list = self.get_sync_command_list(
+            template_file=template_path,
+            code=False,
+            watch=False,
+            resource_id=None,
+            resource=None,
+            dependency_layer=True,
+            stack_name=stack_name,
+            region=None,
+            profile=None,
+            parameter_overrides="Parameter=Clarity",
+            base_dir=None,
+            image_repository=self.ecr_repo_name,
+            image_repositories=None,
+            s3_prefix="integ_deploy",
+            kms_key_id=self.kms_key,
+            capabilities="CAPABILITY_IAM",
+            capabilities_list=None,
+            role_arn=None,
+            notification_arns=None,
+            tags="integ=true clarity=yes foo_bar=baz",
+            metadata=None,
+        )
+
+        sync_process_execute = run_command_with_input(sync_command_list, "y\n".encode())
+        self.assertEqual(sync_process_execute.process.returncode, 1)
+        self.assertIn(
+            "An error occurred (InsufficientCapabilitiesException) when calling the CreateStack operation: \
+Requires capabilities : [CAPABILITY_AUTO_EXPAND]",
+            str(sync_process_execute.stderr),
+        )
+
+    # @parameterized.expand(["template.yaml"])
+    # def test_sync_infra_no_image_repo(self, template_file):
+    #     template_path = str(self.test_data_path.joinpath(template_file))
+    #     stack_name = self._method_to_stack_name(self.id())
+
+    #     # Run infra sync
+    #     sync_command_list = self.get_sync_command_list(
+    #         template_file=template_path,
+    #         code=False,
+    #         watch=False,
+    #         resource_id=None,
+    #         resource=None,
+    #         dependency_layer=True,
+    #         stack_name=stack_name,
+    #         region=None,
+    #         profile=None,
+    #         parameter_overrides="Parameter=Clarity",
+    #         base_dir=None,
+    #         image_repository=None,
+    #         image_repositories=None,
+    #         s3_prefix="integ_deploy",
+    #         kms_key_id=self.kms_key,
+    #         capabilities=None,
+    #         capabilities_list=None,
+    #         role_arn=None,
+    #         notification_arns=None,
+    #         tags="integ=true clarity=yes foo_bar=baz",
+    #         metadata=None,
+    #     )
+
+    #     sync_process_execute = run_command_with_input(sync_command_list, "y\n".encode())
+    #     self.assertEqual(sync_process_execute.process.returncode, 2)
+    #     self.assertIn("", str(sync_process_execute.stderr))
```

Comment:
Same as previous comment, let's clean up redundant/commented code if we are not going to use it.

Reply:
I think I will add a image function test eventually that's why I left this here!